### PR TITLE
ci: unpin peer deps and improve constraints

### PR DESCRIPTION
**Pull request by andreas-karlsson**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing
- [ ] Relevant documentation updated
- [ ] linter/style run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes
- [ ] Tested in a corresponding example app

### CODE REVIEW

File: examples/node/package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
   "dependencies": {
     "@openfeature/core": "^1.1.0",
     "@openfeature/server-sdk": "^1.13.5",
-    "@spotify-confidence/openfeature-server-provider": "0.2.8",
```
Comment (andreas-karlsson) [inline examples/node/package.json:11]: Using these `workspace:` specifiers will make our release PRs a lot less cluttered.

File: packages/openfeature-server-provider/package.json [hunk 2]
```diff
@@ -5,14 +5,15 @@
   "type": "module",
   "types": "build/types/index.d.ts",
   "devDependencies": {
-    "@microsoft/api-extractor": "*",
```
Comment (andreas-karlsson) [inline packages/openfeature-server-provider/package.json:8]: The intention of this star was to use the same version as the workspace root. I don't think that was completely reliable though. If another package depend on for instance `api-extractor` there is nothing saying yarn couldn't pick their version. I've now added this as a specific constraint.

File: packages/openfeature-server-provider/package.json [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
+    "@microsoft/api-extractor": "7.43.1",
     "@openfeature/core": "^1.1.0",
     "@openfeature/server-sdk": "^1.13.5",
-    "rollup": "*"
+    "@spotify-confidence/sdk": "workspace:",
+    "rollup": "4.14.2"
   },
   "peerDependencies": {
     "@openfeature/server-sdk": "^1.13.5",
-    "@spotify-confidence/sdk": "0.1.4"
```
Comment (andreas-karlsson) [inline packages/openfeature-server-provider/package.json:15]: This is the main gist, adding caret to peer-deps. It's also enforced in a constraint.

File: packages/react/package.json [hunk 4]
```diff
@@ -22,14 +22,14 @@
...
   "peerDependencies": {
-    "@spotify-confidence/sdk": "0.1.4",
+    "@spotify-confidence/sdk": "^0.1.4",
     "react": "^18.2.0"
   },
   "devDependencies": {
-    "@microsoft/api-extractor": "*",
-    "@spotify-confidence/sdk": "0.1.4",
+    "@microsoft/api-extractor": "7.43.1",
+    "@spotify-confidence/sdk": "workspace:",
```
Comment (nicklasl) [inline packages/react/package.json:30]:
```suggestion
    "@spotify-confidence/sdk": "0.1.4",
```

**Review by nicklasl — APPROVED** [review]

File: packages/react/package.json (hunk 4 above)
Comment (andreas-karlsson) [reply packages/react/package.json]: Fixed by pinning corresponding dev-deps to lowest possible version.

**Pull request merged**